engine/flink/components/base-tests/src/test/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregatesSpec.scala

```diff
@@ -83,73 +109,209 @@ class AggregatesSpec extends AnyFunSuite with TableDrivenPropertyChecks with Mat
+    }
+  }
+
+  test("should calculate correct results for standard deviation and variance on float") {
+    val table = Table(
+      ("aggregator", "value"),
+      ( SampleStandardDeviationAggregator, Math.sqrt(2.5) ),
+      ( PopulationStandardDeviationAggregator, Math.sqrt(2) ),
+      ( SampleVarianceAggregator, 2.5 ),
+      ( PopulationVarianceAggregator, 2.0 )
+    )
+
+    forAll(table) { (agg, expectedResult) =>
+      val result = addElementsAndComputeResult(List(5.0f, 4.0f, 3.0f, 2.0f, 1.0f), agg)
+      result.asInstanceOf[Double] shouldBe expectedResult +- EPS_DOUBLE
+    }
+  }
+
+  test("should calculate correct results for standard deviation and variance on BigDecimals") {
+    val table = Table(
+      ("aggregator", "value"),
+      ( SampleStandardDeviationAggregator, new java.math.BigDecimal(Math.sqrt(2.5)) ),
+      ( PopulationStandardDeviationAggregator, new java.math.BigDecimal(Math.sqrt(2)) ),
+      ( SampleVarianceAggregator, new java.math.BigDecimal(2.5) ),
+      ( PopulationVarianceAggregator, new java.math.BigDecimal(2.0) )
+    )
+
+    forAll(table) { (agg, expectedResult) =>
+      val result = addElementsAndComputeResult(
+        List(
+          new java.math.BigDecimal("5"),
+          new java.math.BigDecimal("4"),
+          new java.math.BigDecimal("3"),
+          new java.math.BigDecimal("2"),
+          new java.math.BigDecimal("1")
+        ),
+        agg
+      )
+      result
+        .asInstanceOf[java.math.BigDecimal]
+        .subtract(expectedResult)
+        .abs()
+        .compareTo(EPS_BIG_DECIMAL) shouldBe -1
+    }
+  }
+
+  test("should ignore nulls for standard deviation and variance") {
+    val table = Table(
+      ("aggregator", "value"),
+      ( SampleStandardDeviationAggregator, Math.sqrt(2.5) ),
+      ( PopulationStandardDeviationAggregator, Math.sqrt(2) ),
+      ( SampleVarianceAggregator, 2.5 ),
+      ( PopulationVarianceAggregator, 2.0 )
+    )
+
+    forAll(table) { (agg, expectedResult) =>
+      val result = addElementsAndComputeResult(List(null, 5.0, 4.0, null, 3.0, 2.0, 1.0), agg)
+      result.asInstanceOf[Double] shouldBe expectedResult +- EPS_DOUBLE
+    }
+  }
+
+  test("should produce null on empty set") {
+
+    forAll (statisticalAggregators) {agg =>
+      val result = addElementsAndComputeResult(List(), agg)
+      // null is returned because method alignToExpectedType did not run
+      result shouldBe null
+    }
+  }
+
+  test("should calculate correct results for population standard deviation and variance on single element double set") {
+    val table = Table(
+      "aggregator",
+      SampleStandardDeviationAggregator,
+      PopulationStandardDeviationAggregator,
+      SampleVarianceAggregator,
+      PopulationVarianceAggregator
+    )
+    forAll(table) { agg =>
+      val result = addElementsAndComputeResult(List(1.0d), agg)
+      // null is returned because method alignToExpectedType did not run
+      result.asInstanceOf[Double] shouldBe 0
```

Comment:
The comment says that null is returned but in zero was returned? Am I missing sth?